benchmarks/apps/babelstream/babelstream.py

```diff
@@ -0,0 +1,261 @@
+                'Mul':  (0, None, 1555000, 'MBytes/sec'),
+                'Add':  (0, None, 1555000, 'MBytes/sec'),
+                'Triad':  (0, None, 1555000, 'MBytes/sec'),
+                'Dot':  (0, None, 1555000, 'MBytes/sec')
+            },
+# Myriad has four types of GPU nodes: E, F, J and L.
+# L-type nodes each have four NVIDIA 40G A100s. (Compute Capability 80)
+# F-type and E-type nodes each have two NVIDIA Tesla V100s. The CPUs are
+# slightly different on the different letters, see above. (Compute Capability 70)
+# J-type nodes each have two NVIDIA Tesla P100s. (Compute Capability 60)
+            ################################################
+            # 'myriad': {
+            #     'Copy':  (25200, -0.05, 0.05, 'MB/s'),
+            #     'Scale': (16800, -0.05, 0.05, 'MB/s'),
+            #     'Add':   (18500, -0.05, 0.05, 'MB/s'),
+            #     'Triad': (18800, -0.05, 0.05, 'MB/s')
+            # },
+            #################################################
+            'tursa': {
+                'Copy':  (0, None, 600000, 'MB/s'),
+                'Scale': (0, None, 600000, 'MB/s'),
+                'Add':   (0, None, 600000, 'MB/s'),
+                'Triad': (0, None, 600000, 'MB/s')
+            },
+        }
+#--------------------------------------------------
+#-- OpenCL
+#--------------------------------------------------
+@rfm.simple_test
+class OCLBenchmark_CPU(BabelstreamBenchmarkBase):
+    valid_systems = ['-gpu']
+    tags = {"ocl"}
+    executable = "ocl-stream"
+
+@rfm.simple_test
+class OCLBenchmark_GPU(BabelstreamBenchmarkBase):
+    valid_systems = ['+gpu']
+    tags = {"ocl"}
+    executable = "ocl-stream"
+    num_gpus_per_node = 1
+#--------------------------------------------------
+#-- Kokkos
+#--------------------------------------------------
+@rfm.simple_test
+class KOKKOSBenchmark_CPU(BabelstreamBenchmarkBase):
+    valid_systems = ['-gpu']
+    tags = {"kokkos"}
+    executable = "kokkos-stream"
+
+@rfm.simple_test
+class KOKKOSBenchmark_GPU(BabelstreamBenchmarkBase):
+    valid_systems = ['+gpu']
+    tags = {"kokkos"}
+    executable = "kokkos-stream"
+    num_gpus_per_node = 1
+#--------------------------------------------------
+#-- TBB
+#--------------------------------------------------
+@rfm.simple_test
+class TBBBenchmark(BabelstreamBenchmarkBase):
+    valid_systems = ['-gpu']
+    tags = {"tbb"}
+    executable = "tbb-stream"
+    def __init__(self):
+        self.reference = {
+            'isambard-macs:cascadelake': { # https://en.wikichip.org/wiki/intel/xeon_gold/6230https://en.wikichip.org/wiki/intel/xeon_gold/6230 (SINGLE)
+                'Copy':  (0, None, 231860, 'MBytes/sec'),
+                'Mul': (0, None, 231860, 'MBytes/sec'),
+                'Add':   (0, None, 231860, 'MBytes/sec'),
+                'Triad': (0, None, 231860, 'MBytes/sec'),
+                'Dot': (0, None, 231860, 'MBytes/sec')
```

Comment:
Are these reference values correct?  The first element of the tuple should be the value, the second and the third ones the lower and upper bounds, as a fraction of the value if not `None`, but I'm not sure fractions of `0` are useful, but I suspect you wanted to do something like
```py
(231860, None, None, 'MBytes/sec'),
```
instead, although I don't know if we want to have reference values at all here, given previous discussions.